src/scripts/main.js

```diff
@@ -1,3 +1,46 @@
 'use strict';
 
-// write code here
+const parsed = [ ...document.querySelectorAll('li') ];
+
+function sortList(list) {
+  const people = [];
+
+  for (const person in list) {
+    people.push({
+      name: list[person].innerText,
+      position: list[person].dataset.position,
+      salary: +(list[person].dataset.salary).slice(1).replace(/,/g, ''),
+      age: list[person].dataset.age,
+    });
+  }
+
+  const sortedBySalary = [ ...people ].sort((a, b) => +b.salary - +a.salary);
+
+  return sortedBySalary;
+}
+sortList(parsed);
+
+function getEmployees(list) {
+  const people = [];
+
+  for (const person in list) {
+    people.push({
+      name: list[person].innerText,
+      position: list[person].dataset.position,
+      salary: list[person].dataset.salary,
+      age: list[person].dataset.age,
+    });
+  }
+
+  return people;
+}
```

Comment:
If you use `document.querySelector('ul')` you can get the employees list and map it to get the new array of objects without `for` loop.